### DISCUSSION TITLE
Can we use a programmatic configuration in embedded Infinispan Quarkus extension?

### DISCUSSION THREAD

**Discussion by pawissanutt** [infinispan/infinispan#11121] [Q&A] [answered]
Based on the document, the only way to configure Infinispan is by providing an XML file. Is there any workaround to using a programmatic configuration in Quarkus embedded extension? I found the programmatic configuration is more flexible, type-safe, and easier to find the correct configuration by searching through IDE. Right now, I am using Infinispan without extension in my personal project but it would be nice to use Quarkus extension, that enable the native image building.

**Comment by ryanemerson** [comment, marked as answer]
Hi @pawissanutt. It should be possible to use the programmatic configuration with native, however you won't be able to inject the `EmbeddedCacheManager` directly via the producer provided by the extension. Instead you can initialise a `DefaultCacheManager` instance and optionally, expose this via your own Quarkus producer.

**Reply by pawissanutt** [reply]
Thank you for the answer. After some trial and error, I found that  I needed to override the producer provided by the extension. Otherwise, the extension will create its own CacheManager. I have made it works in JVM mode but still not in native mode. I think it might come from other dependencies incompatible with native.
